batchflow/models/eager_torch/layers/resize.py

```diff
@@ -0,0 +1,324 @@
+    def forward(self, x):
+        dim = max(len(get_shape(x)) - 2 - self.dim, 0)
+        ones = [1] * dim
+        return x.view(x.size(0), -1, *ones)
+
+
+class Reshape(nn.Module):
+    """ Enforce desired shape of tensor. """
+    def __init__(self, shape=None, reshape_to=None):
+        super().__init__()
+        self.shape = shape or reshape_to
+
+    def forward(self, x):
+        return x.view(x.size(0), *self.shape)
+
+
+
+class Interpolate(nn.Module):
+    """ Upsample inputs with a given factor.
+
+    Notes
+    -----
+    This is just a wrapper around ``F.interpolate``.
+
+    For brevity ``mode`` can be specified with the first letter only: 'n', 'l', 'b', 't'.
+
+    All the parameters should the specified as keyword arguments (i.e. with names and values).
+    """
+    MODES = {
+        'n': 'nearest',
+        'l': 'linear',
+        'b': 'bilinear',
+        't': 'trilinear',
+    }
+
+    def __init__(self, mode='b', size=None, shape=None, scale_factor=None, **kwargs):
+        super().__init__()
+        self.size, self.scale_factor = size or shape, scale_factor
+
+        if mode in self.MODES:
+            mode = self.MODES[mode]
+        self.mode = mode
+        self.kwargs = kwargs
+
+    def forward(self, x):
+        return F.interpolate(x, mode=self.mode, size=self.size, scale_factor=self.scale_factor,
+                             align_corners=True, **self.kwargs)
+
+    def extra_repr(self):
+        if self.scale_factor is not None:
+            info = 'scale_factor=' + str(self.scale_factor)
+        else:
+            info = 'size=' + str(self.size)
+        info += ', mode=' + self.mode
+        return info
+
+
+class PixelShuffle(nn.PixelShuffle):
+    """ Resize input tensor with depth to space operation. """
+    def __init__(self, upscale_factor=None):
+        super().__init__(upscale_factor)
+
+
+class SubPixelConv(PixelShuffle):
+    """ An alias for PixelShuffle. """
+    pass
+
+
+
+class Crop(nn.Module):
+    """ Crop tensor to desired shape.
+
+    Parameters
+    ----------
+    inputs
+        Input tensor.
+    resize_to : tuple or torch.Tensor
+        Tensor or shape to resize input tensor to.
+    """
+    def __init__(self, resize_to):
+        super().__init__()
+
+        self.resize_to = resize_to
+
+
+    def forward(self, inputs):
+        i_shape = get_shape(inputs)
+        r_shape = get_shape(self.resize_to)
+        if (i_shape[2] > r_shape[2]) or (i_shape[3] > r_shape[3]):
+            # Decrease input tensor's shape by slicing desired shape out of it
+            shape = [slice(None, c) for c in r_shape[2:]]
+            shape = tuple([slice(None, None), slice(None, None)] + shape)
+            output = inputs[shape]
+        elif (i_shape[2] < r_shape[2]) or (i_shape[3] < r_shape[3]):
+            # Increase input tensor's shape by zero padding
+            output = torch.zeros(*i_shape[:2], *r_shape[2:])
+            output[:, :, :i_shape[2], :i_shape[3]] = inputs
+        else:
```

Comment:
If input is greater in one dimension and smaller in other, it will be cropped, but won't be zero-padded